sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SonarAnalyzer.Common.csproj

```diff
@@ -31,13 +33,25 @@
     <AdditionalFiles Include="..\SonarLint.xml" Link="Properties\SonarLint.xml" />
   </ItemGroup>
 
+  <PropertyGroup>
+    <!-- The path to the ucfg submodule, relative to this project -->
+    <UcfgProtoPath>./../../../sonar-ucfg/src/main/protobuf/</UcfgProtoPath>
+  </PropertyGroup>
+
   <ItemGroup>
-    <None Include="..\..\..\sonar-ucfg\src\main\protobuf\ucfg.proto" Link="Protobuf\ucfg.proto" />
+    <None Include="$(UcfgProtoPath)ucfg.proto" Link="Protobuf\ucfg.proto" />
   </ItemGroup>
 
-  <Target Name="GenerateProtobufClasses" BeforeTargets="PrepareForBuild" Outputs="Protobuf\AnalyzerReport.cs,Protobuf\Ucfg.cs" Inputs="Protobuf\AnalyzerReport.proto,..\..\..\sonar-ucfg\src\main\protobuf\ucfg.proto">
-    <Message Importance="high" Text="Generating protobuf class..." />
-    <Exec WorkingDirectory="Protobuf" Command="build.bat" />
+  <!-- Generate protobuf classes, supports windows and linux execution -->
+  <Target Name="GenerateProtobufClasses" BeforeTargets="CoreCompile" Outputs="./Protobuf/Ucfg.cs;.\Protobuf\AnalyzerReport.cs" Inputs="$(UcfgProtoPath)ucfg.proto;.\Protobuf\AnalyzerReport.proto">
```

Comment:
This should also fix the unconditional generation, the Inputs/Outputs separator was `,`, but it should have been `;`